README.md

```diff
@@ -1,60 +1,62 @@
 
-```
-mbed import mbed-os-example-sys-info
-cd mbed-os-example-sys-info
-```
+## Application functionality
 
-### Now compile
+The `main()` function outputs on the serial interface information about the hardware (CPU, ROM and RAM), the firmware (version), as well as the compiler used to build the binary.
 
-Invoke `mbed compile`, and specify the name of your platform and your favorite toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the ARM Compiler 5:
+## Building and Running
 
+1. Connect a USB cable between the USB port on the target and the host computer.
+2. Run the following command to build the example project and program the microcontroller flash memory:
+```bash
+$ mbed compile -m <TARGET> -t <TOOLCHAIN> --flash --sterm
 ```
-mbed compile -m K64F -t ARM
-```
+(Note: Mbed CLI command-line option "--sterm" is used to open serial terminal after compiling.)
 
-Your PC may take a few minutes to compile your code. At the end, you see the following result:
+Your PC may take a few minutes to compile your code.
 
-```
-[snip]
-+------------------+-------+-------+------+
-| Module           | .text | .data | .bss |
-+------------------+-------+-------+------+
-| [lib]\c_w.l      | 11473 |    16 |  348 |
-| [lib]\cpprt_w.l  |    36 |     0 |    0 |
-| [lib]\fz_wm.l    |    18 |     0 |    0 |
-| [lib]\m_wm.l     |    48 |     0 |    0 |
-| anon$$obj.o      |    32 |     0 | 1024 |
-| main.o           |   136 |     0 |    0 |
-| mbed-os\drivers  |   130 |     0 |    0 |
-| mbed-os\features |   132 |     0 |  304 |
-| mbed-os\hal      |  1660 |    30 |   64 |
-| mbed-os\platform |  3465 |   104 |  604 |
-| mbed-os\rtos     | 12926 |  2310 | 4592 |
-| mbed-os\targets  |  9193 |   104 |  324 |
-| Subtotals        | 39249 |  2564 | 7260 |
-+------------------+-------+-------+------+
-Total Static RAM memory (data + bss): 9824 bytes
-Total Flash memory (text + data): 41813 bytes
-
-Image: .\BUILD\K64F\ARM\mbed-os-example-sys-info.bin
-```
+The binary is located at `./BUILD/<TARGET>/<TOOLCHAIN>/mbed-os-example-sys-info.bin` and can alternatively be manually copied to the target which gets mounted on the host computer via USB.
+
+Depending on the target, the example project can be built with GCC_ARM, ARM or IAR toolchain. Run the command below after installing ARM Mbed CLI to determine which toolchain supports your target.
 
-### Program your board
+```bash
+$ mbed compile -S
+```
 
-1. Connect your Mbed device to the computer over USB.
-1. Copy the binary file to the Mbed device.
-1. Press the reset button to start the program.
+## Expected output 
+The serial terminal shows an output similar to the following: 
+``` 
+--- Terminal on /dev/ttyACM0 - 9600,8,N,1 ---
+Mbed OS Version: 51500 
+CPU ID: 0x410fc241 
+Compiler ID: 1 
+Compiler Version: 6130008 
+RAM0: Start 0x20000000 Size: 0x30000 
+RAM1: Start 0x1fff0000 Size: 0x10000 
+ROM0: Start 0x0 Size: 0x100000 
+``` 
+(Note: Compiler IDs are ARM = 1, GCC_ARM = 2, IAR = 3)
```

Comment:
It would be useful to add description for CPUID https://github.com/ARMmbed/mbed-os-example-sys-info/blob/master/main.cpp/#L30-L42 & Compiler version https://github.com/ARMmbed/mbed-os-example-sys-info/blob/master/main.cpp/#L53-L56 as well.